### ISSUE TITLE
List interface supported time stamp doesn't work if using interface index instead of name

### ISSUE THREAD

**Issue by NanXiao** [the-tcpdump-group/tcpdump#769]
Hi,

Greeting from me! On my machine:  
```
$ tcpdump -D
1.docker0 [Up, Running]
2.eno1 [Up, Running]
...
```

If I use interface name to list supported time stamp types, it works OK:  

```
$ tcpdump -J -i eno1
Time stamp types for eno1 (use option -j to set):
  host (Host)
  adapter_unsynced (Adapter, not synced with system time)
```

While using interface index, it doesn't work:  

```
$ tcpdump -J -i 2
Time stamp type cannot be set for 2
```

Do I miss something? Thanks very much!

Best Regards
Nan Xiao

**Comment by stevekay** [comment]
Using the interface number is only for when capturing, per man page.

As no capturing is happening here, the interface number approach doesn't work.

That said, might be a case for an enhancement here, to permit interface numbers being passed to `-J`.
````
       -D
       --list-interfaces
              Print  the  list of the network interfaces available on the system and on which tcpdump can capture packets.  For each network interface, a number and an interface name, possibly
              followed by a text description of the interface, is printed.  The interface name or the number can be supplied to the -i flag to specify an interface on which to capture.

**Comment by fxlb** [comment]
@NanXiao 
There was some change in libpcap a few month ago.
Could you retry with current libpcap/tcpdump ?

**Comment by guyharris** [comment]
> Using the interface number is only for when capturing, per man page.

Then the man page needs to be fixed, because it *is* supposed to work with operations such as listing the link-layer types:

    $ ./tcpdump -L -i 1
    Data link types for en0 when not in monitor mode (use option -y to set):
      EN10MB (Ethernet)
      RAW (Raw IP)

as well as listing the supported time stamp types:

    $ ./tcpdump -J -i 1
    Time stamp types for 1 (use option -j to set):
      host (Host)

If this continues to fail, please report the full output of `tcpdump --version` - and of `tcpdump -D`, because if there is a network interface with the *name* "2", `-j 2` refers to *that* interface, not to the the second interface in the list, at least in newer versions of tcpdump.

**Comment by guyharris** [comment]
> > Using the interface number is only for when capturing, per man page.
> 
> Then the man page needs to be fixed, because it _is_ supposed to work with operations such as listing the link-layer types:

Fixed in f025901fd26f452ea789bb11f3c37c2c84c53c07.

**Comment by NanXiao** [comment]
@fxlb @guyharris Yes, this issue is already fixed, and please close this issue, thanks very much!
